workspaces/arborist/lib/arborist/isolated-reifier.js

```diff
@@ -0,0 +1,449 @@
+const _makeIdealGraph = Symbol('makeIdealGraph')
+const _createIsolatedTree = Symbol.for('createIsolatedTree')
+const _createBundledTree = Symbol('createBundledTree')
+const fs = require('fs')
+const pacote = require('pacote')
+const { join } = require('path')
+const { depth } = require('treeverse')
+const crypto = require('crypto')
+
+module.exports = cls => class IsolatedReifier extends cls {
+  /**
+   * Create an ideal graph.
+   *
+   * An implementation of npm RFC-0042
+   * https://github.com/npm/rfcs/blob/main/accepted/0042-isolated-mode.md
+   *
+   * This entire file should be considered technical debt that will be resolved with an Arborist
+   * refactor or rewrite. Embedded logic in Nodes and Links, and the incremental state of building
+   * trees and reifying contains too many assumptions to do a linked mode properly.
+   *
+   * Instead, this approach takes a tree built from build-ideal-tree, and returns a new tree-like
+   * structure without the embedded logic of Node and Link classes.
+   *
+   * Since the RFC requires leaving the package-lock in place, this approach temporarily replaces
+   * the tree state for a couple of steps of reifying.
+   *
+   **/
+  async [_makeIdealGraph] (options) {
+    /* Make sure that the ideal tree is build as the rest of
+     * the algorithm depends on it.
+     */
+    const bitOpt = {
+      ...options,
+      complete: false,
+    }
+    await this.buildIdealTree(bitOpt)
+    const idealTree = this.idealTree
+
+    function memoize (fn) {
```

Comment:
this function should be declared at module level since it doesn’t need to be created every time the containing function is called.

Reply:
done